packages/lab/src/drawer/Drawer.tsx

```diff
@@ -59,52 +68,69 @@ export const Drawer = forwardRef<HTMLDivElement, DrawerProps>(function Drawer(
   });
 
   const [showComponent, setShowComponent] = useState(false);
+  const { Component: FloatingComponent } = useFloatingComponent();
 
-  const { floating, context } = useDrawer({
+  const id = useId(idProp);
+
+  const { context, floating } = useFloatingUI({
     open,
     onOpenChange,
   });
 
-  const floatingRef = useForkRef<HTMLDivElement>(floating, ref);
+  const { getFloatingProps } = useInteractions([
+    useRole(context, { role: "dialog" }),
+    useClick(context),
+    useDismiss(context),
+  ]);
+
+  const handleRef = useForkRef<HTMLDivElement>(floating, ref);
+
+  const getDrawerProps = () => {
+    return getFloatingProps({
+      ref: floating,
+      id: `${id}-drawer`,
+    });
+  };
 
   useEffect(() => {
     if (open && !showComponent) {
       setShowComponent(true);
     }
-  }, [open, showComponent]);
+  }, [open, showComponent, setShowComponent]);
+
+  if (!showComponent) return <></>;
 
   return (
-    <FloatingPortal>
-      {/* The provider is needed to support the use case where an app has nested modes. The element that is portalled needs to have the same style as the current scope */}
-      <SaltProvider>
-        {showComponent && (
-          <FloatingOverlay className={withBaseName("overlay")} lockScroll>
-            <FloatingFocusManager context={context}>
-              <div
-                ref={floatingRef}
-                className={clsx(
-                  withBaseName(),
-                  withBaseName(position),
-                  {
-                    [withBaseName("enterAnimation")]: open,
-                    [withBaseName("exitAnimation")]: !open,
-                    [withBaseName(variant)]: variant,
-                  },
-                  className
-                )}
-                onAnimationEnd={() => {
-                  if (!open && showComponent) {
-                    setShowComponent(false);
-                  }
-                }}
-                {...rest}
-              >
-                {children}
-              </div>
-            </FloatingFocusManager>
-          </FloatingOverlay>
+    <Scrim>
+      <FloatingComponent
+        aria-modal="true"
+        open={open}
+        ref={handleRef}
+        aria-labelledby={`${id}-header`}
+        aria-describedby={`${id}-content`}
+        focusManagerProps={{
+          context: context,
+        }}
+        className={clsx(
+          withBaseName(),
+          withBaseName(position),
+          {
+            [withBaseName("enterAnimation")]: open,
+            [withBaseName("exitAnimation")]: !open,
+            [withBaseName(variant)]: variant,
+          },
+          className
         )}
-      </SaltProvider>
-    </FloatingPortal>
+        onAnimationEnd={() => {
```

Comment:
What happens if animations are disabled?

Reply:
what's the use case for this - can people disable animations in their browser ?
we could put in a guard that sets open to false after a given time if not already false - or if clicked twice may be a better logic ??
@joshwooding @feliciackh thoughts ?

Reply:
atm if exitAnimation is specifically disabled/removed through classes, the drawer would remain open. Is there a better way to handle it? @joshwooding